pkg/apis/syn/v1alpha1/gitrepo_types.go

```diff
@@ -37,15 +37,15 @@ func (g GitType) IsValid() bool {
 // GitRepoSpec defines the desired state of GitRepo
 type GitRepoSpec struct {
 	// APISecretRef reference to secret containing connection information
-	APISecretRef *corev1.SecretReference `json:"apiSecretRef,omitempty"`
+	APISecretRef *corev1.SecretReference `json:"apiSecretRef"`
 	// DeployKeys optional list of SSH deploy keys. If not set, not deploy keys will be configured
-	DeployKeys []DeployKey `json:"deployKeys,omitempty"`
+	DeployKeys map[string]DeployKey `json:"deployKeys,omitempty"`
 	// Path to Git repository
-	Path string `json:"path,omitempty"`
+	Path string `json:"path"`
 	// RepoName ame of Git repository
-	RepoName string `json:"repoName,omitempty"`
+	RepoName string `json:"repoName"`
 	// TenantRef references the tenant this repo belongs to
-	TenantRef *TenantRef `json:"tenantRef,omitempty"`
+	TenantRef *corev1.LocalObjectReference `json:"tenantRef,omitempty"`
```

Comment:
The problem if we declare this one as mandatory is, that you'll also have to add the tenant ref to the gitRepoTemplate which leads to a very weird user experience when defining clusters. Especially for the cluster where you'll end up defining the same tenantRef twice...

One possible solution for that: we could split the spec into a template and non-template part, where the template doesn't contain the tenant ref and the spec does. That way we can  have both cases.

```suggestion
	// GitRepoSpec defines the desired state of GitRepo
type GitRepoSpec struct {
	GitRepoTemplate
	// TenantRef references the tenant this repo belongs to
	TenantRef corev1.LocalObjectReference `json:"tenantRef"`
}

// GitRepoTemplate is used for templating git repos, it does not contain the tenantRef as it will be added by the
// controller creating the template instance.
type GitRepoTemplate struct {
	// APISecretRef reference to secret containing connection information
	APISecretRef corev1.SecretReference `json:"apiSecretRef"`
	// DeployKeys optional list of SSH deploy keys. If not set, not deploy keys will be configured
	DeployKeys map[string]DeployKey `json:"deployKeys,omitempty"`
	// Path to Git repository
	Path string `json:"path"`
	// RepoName ame of Git repository
	RepoName string `json:"repoName"`
}
```

Reply:
Applied and tested this. It's still a bit weird to handle, but definitely less weird, what do you think?